### PR TITLE
move rate limiting to notify popup itself. add popup for new tlfs

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers

### CODE REVIEW

File: shared/actions/favorite.js [hunk 1]
```diff
@@ -120,6 +124,24 @@ export function favoriteList (): (dispatch: Dispatch) => void {
...
+        const newNotify = {public: folderProps.publicBadge, private: folderProps.privateBadge}
+        const total = folderProps.publicBadge + folderProps.privateBadge
+
+        if (total && !_.isEqual(newNotify, previousNotify)) {
+          previousNotify = newNotify
+
+          let body = `You have ${total} new folders`
+
+          if (total <= 3) {
+            console.log(folderProps.private.tlfs, folderProps.public.tlfs)
```
Comment (chromakode) [inline shared/actions/favorite.js]: Is this `console.log` intended to stay here?

File: shared/actions/favorite.js [hunk 2]
```diff
@@ -120,6 +124,24 @@ export function favoriteList (): (dispatch: Dispatch) => void {
...
+            body = [].concat(folderProps.private.tlfs || [], folderProps.public.tlfs || [])
+              .filter(t => t.meta === 'new')
+              .map(t => t.path).join('\n')
+          }
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: 🐑 🐑 I think it's a little easier to read if the body for more than 3 folders is set in the else statement.

**Comment by MarcoPolo** [comment]
👍

File: shared/actions/favorite.js (hunk 2 above)
Comment (chromakode) [inline shared/actions/favorite.js]: +1

**Comment by chromakode** [comment]
:+1:

File: shared/actions/favorite.js [hunk 3]
```diff
@@ -120,6 +124,25 @@ export function favoriteList (): (dispatch: Dispatch) => void {
...
         dispatch(action)
         dispatch(badgeApp('newTLFs', !!(folderProps.publicBadge || folderProps.privateBadge)))
+
+        const newNotify = {public: folderProps.publicBadge, private: folderProps.privateBadge}
+        const total = folderProps.publicBadge + folderProps.privateBadge
+
+        if (total && !_.isEqual(newNotify, previousNotify)) {
+          previousNotify = newNotify
+
+          let body = null
```
Comment (chromakode) [inline shared/actions/favorite.js]: :sheep: just `let body`?

**Pull request merged**